app/routes/application.js

```diff
@@ -5,9 +5,10 @@ export default class ApplicationRoute extends Route {
   @service session;
   @service router;
   @service userProfile;
+  @service('tasks') tasksService;
 
   async beforeModel() {
-    await this.session.setup();
+    await Promise.all([this.session.setup(), this.tasksService.setup()]);
```

Comment:
Not sure if this is the right place to setup the task service. If the user is not logged in yet, the task list fetched from the backend will be empty and is not refetched after the user authenticated.